### PR TITLE
Not packed integration into the BoolVectorizer

### PR DESCRIPTION
**Pull request by WoutLegiest**
Additional code for not gates into the boolvectorizer
Adding debug code; left to use to build the new version of the vectorizer
Small refactors of function names

### CODE REVIEW

File: lib/Dialect/CGGI/Transforms/BooleanVectorizer.cpp [hunk 1]
```diff
@@ -139,36 +142,58 @@ int bucketSize(const SmallVector<SmallVector<Operation *>> &buckets) {
...
 }
 
+int countNotOps(DenseMap<Operation *, SmallVector<SmallVector<Operation *>>>
+                    compatibleOps) {
+  int count = 0;
+  DenseMap<Operation *, SmallVector<SmallVector<Operation *>>>::iterator it;
+
+  for (it = compatibleOps.begin(); it != compatibleOps.end(); it++) {
+    if (isa<cggi::NotOp>(it->first)) count++;
+  }
```
Comment (j2kun) [inline lib/Dialect/CGGI/Transforms/BooleanVectorizer.cpp:152]: I would try using `llvm::count_if` if possible, then this entire function might be small enough to inline at the sole call site.

https://github.com/llvm/llvm-project/blob/197e0125c3ed991f11d97d27e529c02af06e9f37/llvm/include/llvm/ADT/STLExtras.h#L1945

File: lib/Pipelines/BooleanPipelineRegistration.cpp [hunk 2]
```diff
@@ -156,8 +157,9 @@ void registerTosaToBooleanFpgaTfhePipeline(const std::string &yosysFilesPath,
                                   /*abcBooleanGates=*/true);
 
         // Vectorize CGGI operations
-        pm.addPass(createStraightLineVectorizer(
-            StraightLineVectorizerOptions{.dialect = "cggi"}));
+        // pm.addPass(createStraightLineVectorizer(
+        //     StraightLineVectorizerOptions{.dialect = "cggi"}));
```
Comment (j2kun) [inline lib/Pipelines/BooleanPipelineRegistration.cpp:161]:
```suggestion
```

**Review by j2kun — APPROVED** [review]
LGTM!

File: lib/Dialect/CGGI/Transforms/BooleanVectorizer.cpp (hunk 1 above)
Comment (WoutLegiest) [reply lib/Dialect/CGGI/Transforms/BooleanVectorizer.cpp]: Changed it. 
Cool didn't know that was a thing!

**Pull request merged**